articles/why-data-scientists-need-to-move-from-jupyter-notebooks-to-scripts/index.md

```diff
@@ -0,0 +1,52 @@
+title: Why Data Scientists Need to Move from Jupyter Notebooks to Scripts
+description: Jupyter Notebook is a standard tool in data science because of its ability to allow users to plot and explore data.
+author: eric-kahuha
+date: 2020-09-15T00:00:00-10:00
+topics: []
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/why-data-scientists-need-to-move-from-jupyter-notebooks-to-scripts/hero.jpg
+    alt: Moving from Jupyter to Scripts
+---
+[Jupyter notebook](https://jupyter.org/) is one of the most useful tools for evaluating and exploring data. Data scientists use Jupyter notebook in their daily tasks of data analysis. Interestingly, Jupyter notebook is the first tool that engineers get introduced to when beginning to learn data science.
+<!--more-->
+### Introduction
+Most data science courses rely on this tool as a medium to teach. For beginners, writing code in Jupyter notebook's cells is more comfortable than writing scripts with functions and classes. Besides, Jupyter Notebook is a standard tool in data science because of its ability to allow users to plot and explore data. For instance, when you press Shift + Enter, you expect to see the code's immediate results. This makes it easier for a data engineer to evaluate whether a code snippet is working or not. Despite these benefits, data scientists associate Jupyter Notebook with several setbacks.
+
+### Issues with Jupyter Notebook
+With Jupyter Notebook, you get plenty of vertical screen real estate for visualizing and debugging the data. Which can be a negative, if you find yourself spending a lot of time scrolling through your notebooks to access preliminary analysis and visualizations. You can get lost as your code gets bigger no matter how many markdowns you use to separate the notebook into different sections. Another fallback with the Jupyter Notebook is that it is difficult to experiment with.
+
+A data engineer may want to test different data processing methods by choosing several parameters for his or her machine-learning algorithm to test an increase accuracy. However, every time they experiment with new approaches, they have to find and rerun the associated cells. In the end, these scientists find it time-consuming and confusing, especially when the training or processing procedure takes a long run time.
+
+Moreover, Jupyter Notebook does not support reproducibility. It becomes increasingly difficult to use new data with slightly different structures with Jupyter Notebook. This is because of the challenges involved in identifying the source of the error in the notebook. Not only that, but Jupyter Notebook can make it challenging to debug. In case of an error in a written code, it is difficult to determine whether the reason for the error is the change in data or the code. With Jupyter Notebook not integrating very well with other tools, production is difficult. For example, you cannot run Jupyter Notebook code easily while using other codes.
```

Comment:
For this last sentence: "For example, you cannot run Jupyter Notebook code easily while using other codes."

I can not help but think the audience may be left wondering of a use case example - if you could include (ideal a short snippet) but depending on what you meant -even a short description of an issue or use case - would help the reader imagine what or type of scenario/problem you have in mind.